worker/src/main/java/gov/cms/ab2d/worker/processor/JobPreProcessorImpl.java

```diff
@@ -64,10 +61,16 @@ public Job preprocess(String jobUuid) {
             // If the user provided a 'since' value
             job.setSinceSource(SinceSource.USER);
             jobRepository.save(job);
-        } else if (job.getFhirVersion().supportDefaultSince() && !skipBillablePeriodCheck) {
-            // If the user did not, but this version supports a default 'since', populate it
-            job = updateSinceTime(job);
-            jobRepository.save(job);
+        } else if (job.getFhirVersion().supportDefaultSince()) {
+            boolean hasDateIssue = false;
+            if (job.getContract() != null) {
```

Comment:
How can a job not have a contract?  That should not be possible and should just be asserted which then cleans up the logic.

Reply:
Some of the tests did not include contract numbers. It's not an unreasonable thing to check it.

Reply:
It shouldn't be possible anymore so maybe we need to update the tests in a separate story.

Reply:
Please add a TODO then.  Bonus points for including the ticket number in the comment.